### ISSUE TITLE
uniquelist doesn't work if the (complete) authorlists lists are not unique

### ISSUE THREAD

**Issue by u-fischer** [plk/biblatex#487]
Consider this bib-file where there are two entries for each of the "authorlists":

```
@article {entry1a,
author = {Smith and Johnson and Doe},
title = {Article a},
year = 2005
}

@article {entry1b,
author = {Smith and Johnson and Doe},
title = {Article b},
year = 2005
}

@article {entry2a,
author = {Smith and Koe and Edwards},
title = {Article a},
year = 2005
}

@article {entry2b,
author = {Smith and Koe and Edwards},
title = {Article b},
year = 2005
}
```

If I cite only one entry for each authorlist:

```
\documentclass{article}

\usepackage[
    backend=biber,
    citestyle=authoryear-comp,
    maxcitenames=2,
    uniquelist=true,
    ]
    {biblatex}
\addbibresource{mybib.bib}

\begin{document}
\textcite{entry1a}, \textcite{entry2a}

%\textcite{entry1b}, \textcite{entry2b}

\printbibliography
\end{document}

```

Then the result is as expected with `uniquelist`. The second author is shown too:

![uniquelist1](https://cloud.githubusercontent.com/assets/4047173/18781478/374b721a-8182-11e6-8b42-50c9daaffb2b.PNG)

but if I uncomment the lines with the other two cites then suddenly only one author is shown for every entry:

![uniquelist2](https://cloud.githubusercontent.com/assets/4047173/18781558/941f62bc-8182-11e6-9f25-80ba455c2804.PNG)

I *guess*  that biber tried a uniquelist of 3 for the lists and as it didn't help to distinguish 1a and 1b didn't set a uniquelist value at all instead of using the "second best" value 2. 

This report was triggered by this question:
https://tex.stackexchange.com/questions/330944/setting-uniquelist-true-should-print-second-author-in-all-corresponding-lists-of

**Comment by plk** [comment]
Please try biblatex 3.7 and biber 3.7 from SF, this should be fixed now and regression tests are in place.

**Comment by moewew** [comment]
There was something similar in #336 
```
\documentclass{article}
\usepackage[backend=biber,style=authoryear]{biblatex}

\usepackage{filecontents}
\begin{filecontents}{\jobname.bib}
@article{batty2007iq,
title={IQ in early adulthood and later cancer risk: cohort study of one million Swedish men},
author={Batty, G. David and Wennerstad, Karin Modig and Smith, George Davey and Gunnell, David and Deary, Ian J. and Tynelius, Per and Rasmussen, Finn},
journal={Annals of Oncology},
volume={18},
number={1},
pages={21--28},
year={2007},
publisher={European Society for Medical Oncology}
}

@article{batty2008premorbid,
title={Premorbid (early life) IQ and Later Mortality Risk: Systematic Review},
author={Batty, G. David and Deary, Ian J. and Gottfredson, Linda S.},
journal={Annals of Epidemiology},
volume={17},
number={4},
pages={278--288},
year={2008},
publisher={Elsevier}
}

@article{batty2009iq_early,
title={IQ in Early Adulthood and Mortality By Middle Age: Cohort Study of 1 Million Swedish Men},
author={Batty, G. David and Wennerstad, Karin Modig and Smith, George Davey and Gunnell, David and Deary, Ian J. and Tynelius, Per and Rasmussen, Finn},
journal={Epidemiology},
volume={20},
number={1},
pages={100--109},
year={2009},
publisher={LWW}
}

@article{batty2010iq_socio,
title={IQ in Early Adulthood, Socioeconomic Position, and Unintentional Injury Mortality by Middle Age: A Cohort Study of More Than 1 Million Swedish Men},
author={Batty, G. David and Gale, Catharine R. and Tynelius, Per and Deary, Ian J. and Rasmussen, Finn},
journal={American Journal of Epidemiology},
volume={169},
number={5},
pages={606--615},
year={2010},
publisher={Oxford University Press}
}
\end{filecontents}

\addbibresource{\jobname.bib}

\begin{document}
\begin{refsection}
\section{With \texttt{batty2007iq}}
Possibly ambiguous citations

\autocite{batty2007iq}

\autocite{batty2008premorbid}

\autocite{batty2009iq_early}

\autocite{batty2010iq_socio}
\end{refsection}

\begin{refsection}
\section{Without \texttt{batty2007iq}}
Unambiguous

\autocite{batty2008premorbid}

\autocite{batty2009iq_early}

\autocite{batty2010iq_socio}
\end{refsection}
\end{document} 
```
I still think the output
> <h4>1 With `batty2007iq`: Possibly ambiguous citations</h4>
(Batty et al. 2007)
(Batty, Deary, and Gottfredson 2008)
(Batty et al. 2009)
(Batty, Gale, et al. 2010)
<h4>2 Without `batty2007iq`: Unambiguous</h4>
(Batty, Deary, and Gottfredson 2008)
(Batty, Wennerstad, et al. 2009)
(Batty, Gale, et al. 2010)

is weird.
As far as I can see the fix does not address that issue.

**Comment by plk** [comment]
This should be fixed now in `biber` 2.7 DEV version.
